util-core/src/main/scala/com/twitter/util/Function.scala

```diff
@@ -47,6 +47,38 @@ object Function {
    * Creates `() => A` function from given `Callable`.
    */
   def ofCallable[A](c: Callable[A]): () => A = () => c.call()
+
+  /**
+   * Creates a T => R from a JavaFunction. Used for easier interop
+   * between Java 8 and Twitter Util libraries.
+   */
+  def func[T, R](f: JavaFunction[T, R]) = new Function[T, R] {
+    override def apply(value: T): R = f(value)
+  }
+
+  /**
+   * Creates a T => Unit from a JavaConsumer.
+   * Useful for e.g. future.onSuccess
+   */
+  def cons[T](f: JavaConsumer[T]) = new Function[T, Unit] {
+    override def apply(value: T): Unit = f(value)
+  }
+
+  /**
+   * like func, but deals with checked exceptions as well
+   */
+  def exfunc[T, R](f: ExceptionalJavaFunction[T, R]) = new ExceptionalFunction[T, R] {
```

Comment:
For exactly the same reason why the dispatch library now has explicit operators as well as symbolic ones.  Obscure names are OK for folks who use a library all the time, but just make code harder to read if you only use a library every once in a while.  We do this with our Time and Duration APIs too.

With that said, I don't think this is worth fighting over, and I'd rather get it in, so I'm OK with func.  @mariusae @eirslett are you OK with my original request to camelcase?